Add Net Install ISO doc (#516)
harvester/docs: docs/install/net-install.md

---
sidebar_position: 9
sidebar_label: Net Install ISO
title: "Net Install ISO"
keywords:
  - Harvester
  - Net ISO Installation
  - BMC ISO Redirection
  - BMC Virtual Media

> description: Harvester Net Install ISO is a minimum ISO that contains only the OS binary. It's useful for some situations.

Reviewer: ```suggestion
description: Harvester Net Install ISO is a minimal ISO that contains only the OS binaries. It's useful for some situations.
```

> description: Harvester Net Install ISO is a minimum ISO that contains only the OS binary. It's useful for some situations.

Reviewer: ```suggestion
description: Harvester Net Install ISO is a minimal ISO that contains only the OS binaries. It's useful for some situations.
```

---

<head>
  <link rel="canonical" href="https://docs.harvesterhci.io/v1.3/install/net-install"/>
</head>

> Harvester Net Install ISO is a minimum ISO that contains only the OS binary. It allows booting the Harvester installer to install Harvester OS on a disk. After the OS boots, the Harvester system bootstraps by pulling all required container images from the internet (most from Dockerhub).

Reviewer: ```suggestion
Harvester Net Install ISO is a minimal ISO that contains only the OS binaries. It allows booting the Harvester installer to install Harvester OS on a disk. After the OS boots, the Harvester system bootstraps by pulling all required container images from the internet (most from Dockerhub).
```

A user can use Net Install ISO in these situations:

> - A server has a buggy or slow virtaul media implementation. There are [reports](https://github.com/harvester/harvester/issues/2651) from the community that ISO redirection is too slow to preload all images into a system.

Reviewer: ```suggestion
- A server has a buggy or slow virtual media implementation. There are [reports](https://github.com/harvester/harvester/issues/2651) from the community that ISO redirection is too slow to preload all images into a system.
```

- A user has a private registry containing all Harvester images and has the experience and knowledge to configure image mirrors for containerd.

:::caution

> **A user should always uses the ordinary ISO to bootstrap a Harvester cluster** (e.g., using the ISO without the `-net-install` suffix). The ordinary ISO contains all required images, and the installer preloads all images during the installation. It's easy for a user to reach [Docker Hub's rate limit](https://docs.docker.com/docker-hub/download-rate-limit/) when using a Net Install ISO to bootstrap the Harvester cluster.

Reviewer: I'm not sure about the wording here - "always use the ordinary ISO" kinda implies "never use the Net Install ISO" :-)

Maybe something like "always use the ordinary ISO if possible", or "usually use the ordinary ISO"?

Author: I kind of feel so because it's too easy for a user to reach Docker hub's rate limit :).
I have fixed all the suggested changes, thanks again!

> :::

Reviewer 2: ```suggestion
The Harvester net install ISO is a minimal installation image that contains only the core OS components, allowing the installer to boot and then install the Harvester OS on a disk. After installation is completed, the Harvester OS pulls all required container images from the internet (mostly from Docker Hub).

You can use the net install ISO in the following situations:

- The virtual media implementation on a server is buggy or slow. Community users have reported that ISO redirection is too slow to preload all images onto a system. For more information, see [Issue 2651](https://github.com/harvester/harvester/issues/2651).
- You have a private registry that contains all Harvester images, as well as the knowledge and experience required to configure image mirrors for containerd.

:::caution
**You must always use the full ISO to bootstrap a Harvester cluster** (in other words, use the ISO without the `-net-install` suffix). The full ISO contains all required images, and the installer preloads those images during installation. You can easily reach the [Docker Hub rate limit](https://docs.docker.com/docker-hub/download-rate-limit/) when using a net install ISO to bootstrap the Harvester cluster.
:::
```

> **A user should always uses the ordinary ISO to bootstrap a Harvester cluster if possible** (e.g., using the ISO without the `-net-install` suffix). The ordinary ISO contains all required images, and the installer preloads all images during the installation. It's easy for a user to reach [Docker Hub's rate limit](https://docs.docker.com/docker-hub/download-rate-limit/) when using a Net Install ISO to bootstrap the Harvester cluster.

Reviewer: ```suggestion
**A user should always use the ordinary ISO to bootstrap a Harvester cluster if possible** (e.g., using the ISO without the `-net-install` suffix). The ordinary ISO contains all required images, and the installer preloads all images during the installation. It's easy for a user to reach [Docker Hub's rate limit](https://docs.docker.com/docker-hub/download-rate-limit/) when using a Net Install ISO to bootstrap the Harvester cluster.
```

## Usage

> Download the Net Install ISO from the GitHub release page and boot the ISO to install Harvester. A Net Install ISO has the suffix `net-install` in its name. e.g., https://releases.rancher.com/harvester/v1.3.0/harvester-v1.3.0-amd64-net-install.iso

Reviewer: ```suggestion
Download the Net Install ISO from the [GitHub release page](https://github.com/harvester/harvester/releases) and boot the ISO to install Harvester. A Net Install ISO has the suffix `net-install` in its name. e.g., https://releases.rancher.com/harvester/v1.3.0/harvester-v1.3.0-amd64-net-install.iso
```

> Download the Net Install ISO from the GitHub release page and boot the ISO to install Harvester. A Net Install ISO has the suffix `net-install` in its name. e.g., https://releases.rancher.com/harvester/v1.3.0/harvester-v1.3.0-amd64-net-install.iso

Reviewer 2: ```suggestion
## Usage

Download the net install ISO from the GitHub [Releases](https://github.com/harvester/harvester/releases) page, and then boot the ISO to install Harvester. Net install ISO file names have the suffix `net-install` (for example, https://releases.rancher.com/harvester/v1.3.0/harvester-v1.3.0-amd64-net-install.iso).
```

> ## PXE Installation
> 
> When using the Net Install ISO as the PXE Installation source, please add this kernel parameter when booting the kernel:
> 
> ```
> harvester.install.with_net_images=true
> ```
> 
> Please check [PXE Boot Installation](./pxe-boot-install.md) for more information.

Reviewer 2: ```suggestion
## PXE Installation

If you decide to use the net install ISO as the PXE installation source, add the following parameter when booting the kernel:

```
harvester.install.with_net_images=true
```

For more information, see [PXE Boot Installation](./pxe-boot-install.md).
```

Author: Thanks, updated all suggested changes.

Author: @jillian-maroket Sorry, I think I missed some outdated codes, will add them soon.

Reviewer 2: @bk201 It seems like the wording changes were not applied to `docs/install/net-install.md`.

Author: @jillian-maroket I fixed them up again, thanks!
